src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarFile.cs

```diff
@@ -241,25 +363,112 @@ private static void CreateFromDirectoryInternal(string sourceDirectoryName, Stre
             }
         }
 
+        // Determines what should be the base path for all the entries when creating an archive.
+        private static string GetBasePathForCreateFromDirectory(DirectoryInfo di, bool includeBaseDirectory)
+        {
+            string basePath = di.FullName;
+
+            if (includeBaseDirectory && di.Parent != null)
+            {
+                basePath = di.Parent.FullName;
+            }
+
+            return basePath;
+        }
```

Comment:
it can be one liner?

```suggestion
            => includeBaseDirectory && di.Parent != null ? di.Parent.FullName : di.FullName;
```